pop-api/src/v0/assets/pallets/assets.rs

```diff
@@ -0,0 +1,492 @@
+		delegate: MultiAddress<AccountId, ()>,
+		amount: BalanceParameter,
+	},
+	#[codec(index = 23)]
+	CancelApproval { id: AssetIdParameter, delegate: MultiAddress<AccountId, ()> },
+	#[codec(index = 24)]
+	ForceCancelApproval {
+		id: AssetIdParameter,
+		owner: MultiAddress<AccountId, ()>,
+		delegate: MultiAddress<AccountId, ()>,
+	},
+	#[codec(index = 25)]
+	TransferApproved {
+		id: AssetIdParameter,
+		owner: MultiAddress<AccountId, ()>,
+		destination: MultiAddress<AccountId, ()>,
+		amount: BalanceParameter,
+	},
+	#[codec(index = 26)]
+	Touch { id: AssetIdParameter },
+	#[codec(index = 27)]
+	Refund { id: AssetIdParameter, allow_burn: bool },
+	#[codec(index = 28)]
+	SetMinBalance { id: AssetIdParameter, min_balance: BalanceParameter },
+	#[codec(index = 29)]
+	TouchOther { id: AssetIdParameter, who: MultiAddress<AccountId, ()> },
+	#[codec(index = 30)]
+	RefundOther { id: AssetIdParameter, who: MultiAddress<AccountId, ()> },
+	#[codec(index = 31)]
+	Block { id: AssetIdParameter, who: MultiAddress<AccountId, ()> },
+}
+
+// TODO: Not being used atm but necessary if we want to provide access to the
+//  rest of the pallet, outside of the use cases.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Encode, scale::Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum AssetsError {
+	/// Account balance must be greater than or equal to the transfer amount.
+	BalanceLow,
+	/// The account to alter does not exist.
+	NoAccount,
+	/// The signing account has no permission to do the operation.
+	NoPermission,
+	/// The given asset ID is unknown.
+	Unknown,
+	/// The origin account is frozen.
+	Frozen,
+	/// The asset ID is already taken.
+	InUse,
+	/// Invalid witness data given.
+	BadWitness,
+	/// Minimum balance should be non-zero.
+	MinBalanceZero,
+	/// Unable to increment the consumer reference counters on the account. Either no provider
+	/// reference exists to allow a non-zero balance of a non-self-sufficient asset, or one
+	/// fewer then the maximum number of consumers has been reached.
+	UnavailableConsumer,
+	/// Invalid metadata given.
+	BadMetadata,
+	/// No approval exists that would allow the transfer.
+	Unapproved,
+	/// The source account would not survive the transfer and it needs to stay alive.
+	WouldDie,
+	/// The asset-account already exists.
+	AlreadyExists,
+	/// The asset-account doesn't have an associated deposit.
+	NoDeposit,
+	/// The operation would result in funds being burned.
+	WouldBurn,
+	/// The asset is a live asset and is actively being used. Usually emit for operations such
+	/// as `start_destroy` which require the asset to be in a destroying state.
+	LiveAsset,
+	/// The asset is not live, and likely being destroyed.
+	AssetNotLive,
+	/// The asset status is not the expected status.
+	IncorrectStatus,
+	/// The asset should be frozen before the given operation.
+	NotFrozen,
+	/// Callback action resulted in error.
+	CallbackFailed,
+}
+
+impl TryFrom<u32> for AssetsError {
```

Comment:
Need more of this imo

Reply:
With the current approach this is not necessary anymore. Only if we want to provide a conversion from `Error` to `PalletError`

Reply:
Outdated